packages/gasket-plugin-express/lib/index.d.ts

```diff
@@ -3,19 +3,32 @@ import type { Application, ErrorRequestHandler, Handler } from 'express';
 
 declare module '@gasket/engine' {
   export interface GasketConfig {
-    /** Whether responses are compressed (true by default) */
-    compression?: boolean,
-    /** Filter for which request URLs invoke Gasket middleware */
-    middlewareInclusionRegex?: RegExp,
-    /** Glob pattern for source files setting up express routes */
-    routes?: string,
-    /** @deprecated */
-    excludedRoutesRegex?: RegExp
+    express?: {
+      /** Whether responses are compressed (true by default) */
+      compression?: boolean;
+      /** Filter for which request URLs invoke Gasket middleware */
+      middlewareInclusionRegex?: RegExp;
+      /** Glob pattern for source files setting up express routes */
+      routes?: string;
+      /** @deprecated */
+      excludedRoutesRegex?: RegExp;
+      trustProxy?: boolean | string | number | Function;
+    };
+    middleware?: {
+      plugin: string;
+      paths?: (string | RegExp)[];
+    }[];
   }
 
   export interface HookExecTypes {
-    middleware(app: Application): MaybeAsync<MaybeMultiple<Handler>>,
-    express(app: Application): MaybeAsync<void>,
-    errorMiddleware(app: Application): MaybeAsync<MaybeMultiple<ErrorRequestHandler>>
+    middleware(app: Application): MaybeAsync<MaybeMultiple<Handler> & { paths?: (string | RegExp)[] }>;
```

Comment:
Is this accurate? I'm not seeing anything else in the docs that say that `middleware` can return anything but a middleware function or an array of middleware functions. Is this saying you can have a `paths` property attached to either of those?

Reply:
Is it maybe supposed to be this instead?

```suggestion
    middleware(app: Application): MaybeAsync<MaybeMultiple<Handler & { paths?: (string | RegExp)[] }>>;
```

Reply:
Looks like maybe `paths` can be applied to "layers" which I guess is an array of handlers, so this looks ok after all. It'd be nice if these were documented.

Reply:
Yes I just stumbled onto this code yesterday and had a similar train of thought. I will add some docs.